### DISCUSSION TITLE
Issues after update of VLAD - Embedding training parameters that has worked fine before

### DISCUSSION THREAD

**Discussion by TheRilla** [vladmandic/sdnext#1947] [Q&A] [answered]
Hello there!

I have run into a problem, after I updated to the newest version of VLAD. Earlier i had no problem with training decent embedding, but after I updated the WebUI, I am getting very bad results. Already at around step 600, it looks way overtrained, and I cannot figure out why. I am using the same parameters as I have done before, even tried training the exact same dataset as earlier, but it becomes blurry and full of artifacts.

My settings are:

GeForce 1080Ti

"model_name": "v1-5-pruned-emaonly",

"model_hash": "6ce0161689",

"num_of_dataset_images": 10,

"num_vectors_per_token": 5,

"embedding_name": "TESTemb",

"learn_rate": "0.005:1000, 0.001:2000, 0.0001:5000, 0.00005",

"batch_size": 1,

"gradient_step": 1,

"data_root": "G:\\My Drive\\EMB-images",

"log_directory": "X:\\AI\\VLAD\\automatic\\train\\log\\embeddings\\TESTemb",

"training_width": 512,

"training_height": 512,

"steps": 5000,

"clip_grad_mode": "disabled",

"clip_grad_value": "0.1",

"latent_sampling_method": "deterministic",

"create_image_every": 50,

"save_embedding_every": 50,

"save_image_with_stored_embedding": true,

"template_file": "X:\\AI\\VLAD\\automatic\\train\\templates\\subject_filewords.txt",

"initial_step": 0

I am using a custom subject filewords txt file saying: a photo of [name], [filewords], and the pictures I am using have been interrogated and has worked very well before the update.

Also training settings are set to Scaled-Dot-Product, but I have also tried changing to xFormers, which didn't do anything. I've also made sure I have no VAE loaded.

I've tried to check out the guide on training embeddings on StableDiffusion subreddit since it helped me a lot in the past, but I cannot see what I am doing wrong, except for not being able to find " Use BLIP for caption " setting. To be honest, I don't remember if I used it in the past.

I hope you can help me.

**Comment by vladmandic** [comment, marked as answer]
`latent_sampling_method=deterministic` was broken pretty much forever (not in sdnext, this is going back to first a1111 ti training), i've recently fixed it so it actually does what its supposed to do, but as a result you're getting overfitted results.
either dont use deterministic or change other params. in general, i really don't know who/what/why deterministic became popular, its never really a good idea as it does lead to exactly what you see - overfitting. only when deterministic is not working, you're getting good results.

**Reply by TheRilla** [reply]
Thank you for your assistance sir, this seems to have fixed the issue!
